File: gitdot-web/packages/api/src/endpoint/user/get-current-user-settings.ts
import { z } from "zod";
import { UserSettingsResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const GetCurrentUserSettingsRequest = z.object({});
export type GetCurrentUserSettingsRequest = z.infer<
  typeof GetCurrentUserSettingsRequest
>;

export const GetCurrentUserSettingsResponse = UserSettingsResource;
export type GetCurrentUserSettingsResponse = z.infer<
  typeof GetCurrentUserSettingsResponse
>;

export const GetCurrentUserSettings = {
  path: "/user/settings",
  method: "GET",
  request: GetCurrentUserSettingsRequest,
  response: GetCurrentUserSettingsResponse,
} as const satisfies Endpoint;
export type GetCurrentUserSettings = typeof GetCurrentUserSettings;

File: gitdot-api/src/endpoint/user/get_current_user_settings.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::user::UserSettingsResource};

pub struct GetCurrentUserSettings;

impl Endpoint for GetCurrentUserSettings {
    const PATH: &'static str = "/user/settings";
    const METHOD: http::Method = http::Method::GET;

    type Request = GetCurrentUserSettingsRequest;
    type Response = GetCurrentUserSettingsResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct GetCurrentUserSettingsRequest {}

pub type GetCurrentUserSettingsResponse = UserSettingsResource;
